### ISSUE TITLE
Large Training Error when use descriptor: se_e3 and activation_function: gelu

### ISSUE THREAD

**Issue by swome** [deepmodeling/deepmd-kit#1332]
Training Errors are very large when I use both descriptor: se_e3 and activation_function: gelu in **EMBEDDING NET**; their is NO bug  when using activation_function: gelu in **FITTING NET**

Version: v2.0.1, commit 2f6020b, running on Lebesgue.

{ "_comment": "that's all",
    "model": {
        "type_map": [
            "Li"
        ],
        "descriptor": {
            "type": "hybrid",
            "list": [
                {
                    "type": "se_a",
                    "sel": [
                        250
                    ],
                    "rcut_smth": 0.0,
                    "rcut": 6.0,
                    "neuron": [
                        20,
                        40,
                        80
                    ],
                    "resnet_dt": false,
                    "axis_neuron": 16,
                    "_comment": " that's all",
                    "seed": 2033265099,
                    "activation_function": "gelu"
                },
                {
                    "type": "se_at",
                    "sel": [
                        35
                    ],
                    "rcut_smth": 0.0,
                    "rcut": 3.0,
                    "neuron": [
                        4,
                        8,
                        16
                    ],
                    "resnet_dt": false,
                    "_comment": " that's all",
                    "seed": 3445690150,
                    "activation_function": "tanh"
                }
            ]
        },
        "fitting_net": {
            "neuron": [
                240,
                240,
                240
            ],
            "resnet_dt": true,
            "_comment": " that's all",
            "seed": 2715686539,
            "activation_function": "tanh"
        },
        "_comment": " that's all"
    },
    "learning_rate": {
        "type": "exp",
        "start_lr": 0.001,
        "stop_lr": 3.51e-08,
        "decay_steps": 5000,
        "_comment": "that's all"
    },
    "loss": {
        "start_pref_e": 0.2,
        "limit_pref_e": 1,
        "start_pref_f": 100,
        "limit_pref_f": 1,
        "start_pref_v": 0.0,
        "limit_pref_v": 0,
        "_comment": " that's all"
    },
    "training": {
        "stop_batch": 1000000,
        "_comment": "that's all",
        "disp_file": "lcurve.out",
        "disp_freq": 1000,
        "numb_test": 5,
        "save_freq": 1000,
        "save_ckpt": "model.ckpt",
        "disp_training": true,
        "time_training": true,
        "profiling": false,
        "profiling_file": "timeline.json",
        "training_data": {
            "systems":'"../"
            "batch_size": "auto"
            "auto_prob": "prob_sys_size; 0:197:0.900000; 197:200:0.100000"
        },
        "seed": 982186638
    }
}

Resultant lcurve
 991000      2.32e+00      3.35e-01      2.17e-02    3.9e-08
 992000      5.18e-01      6.13e-02      2.96e-01    3.9e-08
 993000      2.13e+00      3.20e-01      6.24e-02    3.9e-08
 994000      4.96e-01      1.22e-01      8.37e-02    3.9e-08
 995000      8.62e-01      1.24e-01      4.82e-02    3.7e-08
 996000      1.11e+00      1.51e-01      3.50e-01    3.7e-08
 997000      1.02e+00      2.25e-01      1.78e-01    3.7e-08
 998000      9.68e-01      1.23e-01      4.55e-01    3.7e-08
 999000      3.55e-01      2.91e-02      2.92e-01    3.7e-08
1000000      2.94e+00      4.43e-01      5.86e-02    3.5e-08

Gelu seems not to work well with se3

**Comment by njzjz** [comment]
> "type": "se_a",

The parameters you provided set `type` to `se_a`, but not `se_e3`. Could you please confirm it?

**Comment by swome** [comment]
it is a hybrid of se_a and se_e3.

"type": "hybrid",
"type": "se_a",
"type": "se_at",

Jinzhe Zeng ***@***.***> 于2021年12月4日周六 17:31写道：

**Comment by njzjz** [comment]
You only set the activation function of `se_a` to gelu. The activation function of `se_e3` is still tanh.

**Comment by swome** [comment]
Sorry, I pasted the wrong script. The script that reproduces the bug looks
like this:
[image: image.png]
and the lcurve is like:
[image: image.png]





Jinzhe Zeng ***@***.***> 于2021年12月5日周日 10:33写道：

**Comment by njzjz** [comment]
Your images are not visible.

**Comment by njzjz** [comment]
Close as there is no enough information to reproduce the problem.
